flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunner.java

```diff
@@ -392,7 +392,7 @@ private void forwardResultFuture(
                         if (isValidLeader(leaderSessionId)) {
                             onJobCompletion(jobManagerRunnerResult, throwable);
                         } else {
-                            LOG.trace(
+                            LOG.debug(
```

Comment:
The entire callback can be replaced by `runIfValidLeader`.

Reply:
These codes have been replaced by `runIfValidLeader`.

Reply:
You missed `forwardIfValidLeader`

Reply:
You missed `forwardIfValidLeader`.